src/components/Avatar.js

```diff
@@ -70,6 +72,16 @@ function Avatar(props) {
         return null;
     }
 
+    useEffect(() => {
+        if (!_.isFunction(props.source) && props.network.isOffline) {
+            const cachedImage = new Image();
+            cachedImage.src = props.source;
+            if (!cachedImage.complete) {
+                setImageError(true);
+            }
+        }
+    }, []);
```

Comment:
Remove that. We don't need this kind of verification. Just change the initial value of `imageError` from `false` to `isOffline` where `isOffline` is defined from the network hook

```js
const {isOffline} = useNetwork();
const [imageError, setImageError] = useState(isOffline);
```


BTW, `Image` is web only.

Reply:
If we set the imageError as True (isOffline) initially.
https://github.com/Expensify/App/assets/51104065/0cd8f324-0b3c-4a87-9ecd-dfcc79796de2

Reply:
So I used useEffect.

Reply:
How about this?
```
useEffect(() => {
  if (_.isFunction(props.source) || !isOffline) return;
  setImageError(true);
}, [])
```

Reply:
Oh I see, the images are no longer loaded from cache if offline and just use the fallback image. I think we should fix that (or confirm the expected behaviour with the team on slack)

any idea here on how we can fix this?

Reply:
We told the above comments If we cannot use Image (only web), no ideas for now.

We can also agree with not using cached avatars while offline, but pattern avatars.

Reply:
I can't find your Slack, here is the thread https://expensify.slack.com/archives/C01GTK53T8Q/p1688513730916269 asking for the expected behaviour

Reply:
@s77rt 
I tested with [this code](https://github.com/Expensify/App/pull/22229#discussion_r1252399665), it seems much better from now.

Reply:
How is that much better?

Reply:
I think we should fix the cache for other cloud images as well

Reply:
@s77rt Can we check cache algorithm only on the web?
I am not sure this issue is also occurred in mobile.

Reply:
We should not do that. That algo seems as a hack

Reply:
Then let's go with https://github.com/Expensify/App/pull/22229#discussion_r1252399665
I think this is the best option for now.
I investigated all solutions, but there are no fantastic ways.

Reply:
Unfortunately that solution won't fix the cache problem.